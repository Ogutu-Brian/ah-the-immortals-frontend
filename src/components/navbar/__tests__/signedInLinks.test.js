import React from 'react';
import { Router, NavLink } from 'react-router-dom';
import { mount } from 'enzyme';
import { createBrowserHistory } from 'history';
import toJson from 'enzyme-to-json';
import SignedInLinks from '../signedInLinks';

describe('Tests rendering of links on navigation bar when the user is logged in', () => {
  const history = createBrowserHistory();
  const mockHandleLogout = jest.fn;
  const mockHandleClick = jest.fn;
  const mockHandleClear = jest.fn;
  const username = '';
  const notifications = [
    {
      'id': 13,
      'unread': false,
      'verb': 'article_created',
      'timestamp': '2019-06-19 11:45:49',
      'description': 'barclay.koin posted an article "TESTING FOR GOT" on 19-June-2019 11:45'
    }];
  const unreadNotifications = [
    {
      'id': 10,
      'unread': false,
      'verb': 'article_created',
      'timestamp': '2019-06-19 11:45:49',
      'description': 'barclay.koin posted an article "TESTING FOR GOT" on 19-June-2019 11:45'
    },
    {
      'id': 9,
      'unread': false,
      'verb': 'article_created',
      'timestamp': '2019-06-19 11:45:49',
      'description': 'barclay.koin posted an article "TESTING FOR GOT" on 19-June-2019 11:45'
    }
  ];
  const wrapper = mount(
    <Router history={history}>
      <SignedInLinks 
        handleLogout={mockHandleLogout}
        username={username}
        handleClick={mockHandleClick}
        notifications={notifications}
        unread={unreadNotifications.length}
        handleClear={mockHandleClear}
      />
    </Router>

  );
  it('renders signed in links component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('handle logout', () => {
    const logoutBtn = wrapper.findWhere(n => n.type() === NavLink && n.contains('Logout'));
    expect(logoutBtn.length).toEqual(1);
  });
});
