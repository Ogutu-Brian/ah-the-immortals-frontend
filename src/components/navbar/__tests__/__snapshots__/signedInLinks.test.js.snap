// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests rendering of links on navigation bar when the user is logged in renders signed in links component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <SignedInLinks
    handleClear={[Function]}
    handleClick={[Function]}
    handleLogout={[Function]}
    notifications={
      Array [
        Object {
          "description": "barclay.koin posted an article \\"TESTING FOR GOT\\" on 19-June-2019 11:45",
          "id": 13,
          "timestamp": "2019-06-19 11:45:49",
          "unread": false,
          "verb": "article_created",
        },
      ]
    }
    unread={2}
    username=""
  >
    <div
      className="nav ml-auto"
    >
      <div
        className="bell"
      >
        <p
          className="notification"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            className="button-bell"
            data-target="#notificationsmenu"
            data-toggle="dropdown"
            id="note-bell"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fa fa-bell notnotif text-dark mt-3"
            />
          </button>
          <span
            className="badge"
          >
            2
          </span>
        </p>
        <div
          className="dropdown-menu  dropdown-menu-sm-left scrollable-menu"
          id="notificationsmenu"
        >
          <p
            className="notes"
          >
            Notifications
            <button
              className="clear-note"
              id="clear-btn"
              onClick={[Function]}
              type="button"
            >
              Clear All
            </button>
          </p>
          <div
            className="dropdown-divider"
          />
          <ul
            className="text-left notify"
            key="13"
          >
            <li
              className="dropdown-item menu-item"
            >
              <Link
                to="/post/article_created"
              >
                <a
                  href="/post/article_created"
                  onClick={[Function]}
                >
                  barclay.koin posted an article "TESTING FOR GOT" on 19-June-2019 11:45
                </a>
              </Link>
            </li>
          </ul>
        </div>
      </div>
      <NavLink
        className="nav-link mt-1 mr-1"
        to="/postarticle"
      >
        <Link
          aria-current={null}
          className="nav-link mt-1 mr-1"
          to="/postarticle"
        >
          <a
            aria-current={null}
            className="nav-link mt-1 mr-1"
            href="/postarticle"
            onClick={[Function]}
          >
            Post Article
          </a>
        </Link>
      </NavLink>
      <NavLink
        aria-expanded="true"
        className="nav-link dropdown-toggle"
        data-target="#profileDropdown"
        data-toggle="dropdown"
        role="button"
        to=""
      >
        <Link
          aria-current={null}
          aria-expanded="true"
          className="nav-link dropdown-toggle"
          data-target="#profileDropdown"
          data-toggle="dropdown"
          role="button"
          to=""
        >
          <a
            aria-current={null}
            aria-expanded="true"
            className="nav-link dropdown-toggle"
            data-target="#profileDropdown"
            data-toggle="dropdown"
            href="/"
            onClick={[Function]}
            role="button"
          >
            <i
              className="fa fa-user-circle ml-3 mr-2 mt-2"
            />
            <span
              className="mr-1"
            />
          </a>
        </Link>
      </NavLink>
      <div
        className="dropdown-menu ml-5"
        id="profileDropdown"
      >
        <div
          className="dropdown-item"
        >
          <NavLink
            className="nav-link"
            to="/myProfile/"
          >
            <Link
              aria-current={null}
              className="nav-link"
              to="/myProfile/"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/myProfile/"
                onClick={[Function]}
              >
                Profile
              </a>
            </Link>
          </NavLink>
        </div>
        <div
          className="dropdown-item"
        >
          <NavLink
            className="nav-link"
            to="/"
          >
            <Link
              aria-current="page"
              className="nav-link active"
              style={Object {}}
              to="/"
            >
              <a
                aria-current="page"
                className="nav-link active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                Settings
              </a>
            </Link>
          </NavLink>
        </div>
        <div
          className="dropdown-item"
        >
          <NavLink
            className="nav-link"
            onClick={[Function]}
            to="/"
          >
            <Link
              aria-current="page"
              className="nav-link active"
              onClick={[Function]}
              style={Object {}}
              to="/"
            >
              <a
                aria-current="page"
                className="nav-link active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                Logout
              </a>
            </Link>
          </NavLink>
        </div>
      </div>
    </div>
  </SignedInLinks>
</Router>
`;
